hooks/tk-multi-publish2/basic/publish_session_geometry.py

```diff
@@ -0,0 +1,514 @@
+                it will be hidden. Optional, True by default.
+            - checked: If True, the plugin will be checked in the UI, otherwise
+                it will be unchecked. Optional, True by default.
+
+        :param settings: Dictionary of Settings. The keys are strings, matching
+            the keys returned in the settings property. The values are `Setting`
+            instances.
+        :param item: Item to process
+
+        :returns: dictionary with boolean keys accepted, required and enabled
+        """
+
+        accepted = True
+        publish_template = self._get_template("Publish file Template", settings)
+        if not publish_template and not item.parent.properties["work_file_template"]:
+            accepted = False
+
+        # check that the AbcExport command is available!
+        if not mel.eval("exists \"AbcExport\""):
+            accepted = False
+            self.logger.warn("Item not accepted because export command AbcExport is not available.")
+        
+        return {
+            "accepted": accepted,
+            "checked": True
+        }
+
+    def validate(self, settings, item):
+        """
+        Validates the given item to check that it is ok to publish. Returns a
+        boolean to indicate validity.
+
+        :param settings: Dictionary of Settings. The keys are strings, matching
+            the keys returned in the settings property. The values are `Setting`
+            instances.
+        :param item: Item to process
+        :returns: True if item is valid, False otherwise.
+        """
+
+        # check that there is still geometry in the scene:
+        if not cmds.ls(geometry=True, noIntermediate=True):
+            self.logger.warn("Validation failed because there is no geometry in the scene to be exported")
+            return False
+
+        publisher = self.parent
+
+        # ensure we have an updated project root
+        project_root = cmds.workspace(q=True, rootDirectory=True)
+        item.properties["project_root"] = project_root
+
+        # get the configured work file template
+        work_template = item.parent.properties["work_file_template"]
+        publish_template = self._get_template("Publish file Template", settings)
+
+        # get the current scene path and extract fields from it
+        # using the work template:
+        scene_path = os.path.abspath(cmds.file(query=True, sn=True))
+        #scene_file_name = os.path.basename(scene_path)
+        scene_file_name = publisher.util.get_publish_name(scene_path)
+        scene_file_name_root, scene_file_name_ext = os.path.splitext(scene_file_name)
+        fields = work_template.get_fields(scene_path) #{ "asset_root": project_root, "name": scene_file_name_root, "version": 0 }
+                
+        # create the publish path by applying the fields 
+        # with the publish template:
+        publish_path = publish_template.apply_fields(fields)
```

Comment:
I wonder if we should use the `missing_fields` method on the publish template to make sure we have the required fields. https://github.com/shotgunsoftware/tk-core/blob/master/python/tank/template.py#L154. We're not doing this in the other hooks, but maybe we should. Let's discuss on Monday.

Reply:
Pending for discussion...